src/app/ui/user-profile/user-profile.component.ts

```diff
@@ -1,17 +1,76 @@
 import { Component } from '@angular/core';
-
 import { AuthService } from '../../core/auth.service';
+import { ActivatedRoute } from '@angular/router';
+import { UserService } from '../../core/user.service';
+import { MatDialog } from '@angular/material/dialog';
+import { AddQuestDialogComponent } from '../dialog/add-quest-dialog/add-quest-dialog.component';
+import { SeasonService } from '../../core/season.service';
 
 @Component({
   selector: 'user-profile',
   templateUrl: './user-profile.component.html',
   styleUrls: ['./user-profile.component.scss'],
 })
 export class UserProfileComponent {
+  private player: User;
+  private userServiceSub: any;
+  private routeSub: any;
+  private user: User;
+  private userSub: any;
+  private season: Season;
+  private seasonSub: any;
 
-  constructor(public auth: AuthService) { }
+  constructor(private auth: AuthService, 
+    private route: ActivatedRoute,
+    private userService: UserService,
+    private dialog: MatDialog,
+    private seasonService: SeasonService) { }
 
   logout() {
     this.auth.signOut();
   }
-}
+
+  ngOnInit() {
+    //get the logged-in user data
+    this.userSub = this.auth.user.subscribe((user: User) => {
+      this.user = user;
+
+      //check if the logged-in user is a team lead 
+      if(this.user.isLead) {
+        //get the player id from the path
+        this.routeSub = this.route.params.subscribe(param => {
+          var playerId = param['playerId'];
+
+          //this will tell us if the logged-in user is viewing his own data, or is viewing someone else's
+          if(playerId) {
+            //get the details of the player being viewed by the logged-in user
+            this.userServiceSub = this.userService.getUser(playerId).subscribe((player: User) => {
+              this.player = player;
+
+              this.seasonSub = this.seasonService.getEnabledSeason().subscribe((season: Season) => {
```

Comment:
i think this is not dependent at all on the user info. 
and also to prevent using stale seasonId (where enabled season changes after you've get the id), we need to do this just before the transaction where it is really needed.

Reply:
the team lead is assigning the quest to the user, the current used data loaded is the team lead's data. For the player's quest creation, data needed is the specific user's details.